### DISCUSSION TITLE
Integration of Glow as a TUI Task Dashboard

### DISCUSSION THREAD

**Discussion by ProfessorManhattan** [go-task/task#1077] [Show and tell]
Hey, I have created a couple different forks of Task. One of them integrates the Glow TUI menu so that when you run `task --menu` it shows a very pretty TUI menu that helps developers navigate through tasks easier --- the main goal is to help developers onto a project quicker.

Here's the fork: https://github.com/megabyte-labs/task-dash
Here's an issue where we are discussing the possibility of merging this into the upstream project: https://github.com/megabyte-labs/task-dash/issues/1
Another fork with a bunch of new features (documented in the README): https://github.com/megabyte-labs/bodega
YouTube walkthrough that might be helpful if you want to test the feature (also runs through some of the issues with it): https://github.com/megabyte-labs/bodega

Thunderstruck started playing while posting this - https://www.cnet.com/culture/thunderstruck-a-tale-of-malware-acdc-and-irans-nukes/

**Comment by ProfessorManhattan** [comment]
Hey @gedw99 --- wow... the JSON idea is great!! I wanted to make a GUI so I actually had someone play with opening a websocket but maybe another library can handle the connection and just get the data from `--json`. I think some of the code for the websocket might be in the Bodega fork.

**Reply by ProfessorManhattan** [reply]
Yup --- I see `--server            runs as a server` in the README for Bodega

**Comment by ProfessorManhattan** [comment]
Yeah, the merge would have to be 🍒 picked to only include the go
modifications. Hopefully I didn't run an auto formatter on the code 😬

**Comment by ghostsquad** [comment]
@ProfessorManhattan could you update the description with the link to the YouTube video? It looks like there was an erroneous double paste of the github repo link.

**Comment by andreynering** [comment]
Hi @ProfessorManhattan,

Sorry for taking a bit to respond to this.

I don't think I have a strong opinion at the moment on whether we should bring this TUI into Task or not.

For one side, a simple TUI could be useful and I think many users would like that. On the other side, every feature we bring into the codebase needs to be maintained, and I'm not sure how much code that means or how complex it is.

So, some thoughts:

1) If easy enough, you can consider opening a PR with the code so we can take a look to try to understand better how that works. If that's not too easy, consider just pointing me into the right files in your codebase instead, so you don't have unnecessary work as there's no guarantee we'd merge that.
2) Instead of maintaining a whole fork of Task, you may want to move your project to use the new `--json` flag. That's already being used by the VSCode extension, and I believe it could be easier to maintain compared to a fork. That looks like a good alternative to me: a community project that's just an additional binary (`task-tui` or `task-menu`, for example).

I think I'm currently inclined to prefer option 2, maintained by the community. This way we (the main maintainers) can keep focused on the core tool, which is already a significant amount of work.
